### PR TITLE
Awesome books with es6

### PR DESCRIPTION
**Pull request by gnuion**
In this pull request I:
- Create single page application
- Use JavaScript modules
- Use local storage
- Add routing
- Use luxon library for date and time

### CODE REVIEW

File: modules/bookList.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/* eslint-disable */
```
Comment (vechicin) [inline modules/bookList.js:1]: Please do not disable ESLint.

File: modules/date.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/* eslint-disable */
```
Comment (vechicin) [inline modules/date.js:1]: Please do not disable ESLint.

File: modules/router.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+/* eslint-disable */
```
Comment (vechicin) [inline modules/router.js:1]: Please do not disable ESLint.

**Review by vechicin — CHANGES_REQUESTED** [review]
# Hi @gnuion ✋, 

Good job so far! 🔥 
There are some issues that you still need to work on to go to the next project but you are almost there! 🚨 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights ⚡ 

- [x] Good job using ES6 features and modularizing your project properly!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 📂 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gnuion** [comment]
JavaScript is an old language, and the linters complain about personal preferences. Inside date.js they complain about using export default instead of export. I know I have only one function there, but I might decide to add more in the future.
image
 

In other places, it complains about a trailing comma at the end of a function call. But I don't want to add one. Prettier is okay with it.

image


And sometimes I don't want to use template literals, because it adds complexity I don't want with `${}`. 
image


I hate JavaScript because Dart is so much better 💙!

**Review by nicupop729 — COMMENTED** [review]
Hello @gnuion  👋,

_This is an additional review due to the second opinion request._

Linters are great for identifying errors when you use standard rules. Remember, a linter analyzes your code for stylistic and programming errors against the rules it knows. If part of your code breaks the standard rules, this can pose a problem.
At Microverse we are using linters for whole students to follow the same standard rules.
You are free to use Dart for your side project, and please do so. But because here we are using JavaScript, you have as well to follow the rules.
The code reviewer, @vechicin, was right when he asked you to enable back the linters, and maybe you just needed a bit more explaining from his side.

**To progress with this project, enable back the linters, correct the potential errors and ask again for a review in your dashboard.**

(As an exception, if Luxon gives a lot of errors, you are allowed to disable linters in that particular file only).

Keep rocking 🚀!

**Comment by gnuion** [comment]
> Hello @gnuion 👋,
> 
> _This is an additional review due to the second opinion request._
> 
> Linters are great for identifying errors when you use standard rules. Remember, a linter analyzes your code for stylistic and programming errors against the rules it knows. If part of your code breaks the standard rules, this can pose a problem. At Microverse we are using linters for whole students to follow the same standard rules. You are free to use Dart for your side project, and please do so. But because here we are using JavaScript, you have as well to follow the rules. The code reviewer, @vechicin, was right when he asked you to enable back the linters, and maybe you just needed a bit more explaining from his side. Another important aspect is that this project is supposed to be done in a pair-programming activity. The project should have been invalidated straight away by the previous reviewer, @vechicin, because of this aspect.
> 
> **To progress with this project, you have to work together in a team (commits from your partner must appear in the thread, as this project must be done in one week time together with your partner), enable back the linters, correct the potential errors and ask again for a review in your dashboard.**
> 
> (As an exception, if Luxon gives a lot of errors, you are allowed to disable linters in that particular file only).
> 
> Keep rocking 🚀!

TypeScript doesn't need linters. Vanilla JavaScript is stupid. I hate that concept of falsy and truthy. Hopefully with WASM JavaScript will die. Until then, I have to live with it. And this is not a pair programming exercise. OMG!

image

**Comment by nicupop729** [comment]
Hi @gnuion,
Regarding the pair-programming activity, you are right, it was my mistake, it is just a solo activity.

Please go ahead with the other requirement, which is enabling back the linters and when you are ready, please ask for another review.

Keep rocking! :rocket:

**Comment by gnuion** [comment]
@vechicin @nicupop729 I just wanted to raise a discussion here, I don't think that your reviews are bad. I just don't like that trailing comma error. If so, we should implement trailing commas everywhere, also in arrays and objects. Doesn't make sense to me. Anyhow, this is JavaScript, so what do I except 😂. Thank you!

**Review by mukhammadrizooff — APPROVED** [review]
Hi @gnuion, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter errors ✔️ 
- Good implementations ✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**